frontend/src/todos/todosList.tsx

```diff
@@ -104,57 +104,65 @@ function TodosList() {
       setTodos({ status: 'failed', error: error.message });
     }
   }
+
+  function Spinner(){
+    return(
+      <div
+      style={{
+        position: 'absolute',
+        top: '50%',
+        left: '50%',
+        transform: 'translate(-50%, -50%)',
+        zIndex: 1000,
+      }}
+    >
+      <RingLoader color=" #36ccae" size={150} />
+    </div>
+    )
+  }
+
+  function Body(prpos: {isPending:boolean, todos: Todo[]}){
+    return(
+      <>
+        {prpos.isPending && Spinner()}
+      <div
+        style={{
+          pointerEvents: prpos.isPending ? 'none' : 'auto',
+          opacity: prpos.isPending ? 0.5 : 1,
+        }}
+      >
+        <TodoInput
+          handleAddTodo={handleAddTodo}
+          handleDeleteAll={handleDeleteAll}
+        />
+        <div className="listsOfTodos">
+          <TodosListPresenter
+            todos={prpos.todos}
+            isFinished={false}
+            onToggle={handleToggle}
+            onDelete={handleDelete}
+          />
+          <TodosListPresenter
+            todos={prpos.todos}
+            isFinished={true}
+            onToggle={handleToggle}
+            onDelete={handleDelete}
+          />
+        </div>
+      </div>
+    </>
+    )
+  }
+
   switch (todos.status) {
     case 'idle':
       return null;
     case 'failed':
       return <div>Error: {todos.error}</div>;
     case 'loaded':
+      return Body({isPending: false, todos: todos.data})
```

Comment:
components should always be created using <Body... > if you want to return a jsx without a component
then the function should be named with small letter ("body" and not "Body")